cmake/CMakeLists.txt

```diff
@@ -1172,14 +1172,9 @@ if (onnxruntime_ENABLE_TRAINING)
           message(STATUS "NCCL_MINOR_VERSION: ${NCCL_MINOR_VERSION}")
         endif()
 
-        if (${NCCL_MAJOR_VERSION} GREATER 2)
+        if ("${NCCL_MAJOR_VERSION}.${NCCL_MINOR_VERSION}" VERSION_GREATER_EQUAL "2.7")
```

Comment:
should we check if NCCL_MAJOR_VERSION/NCCL_MINOR_VERSION contains a value?

Reply:
hm, it may be fine. might be worth testing out the behavior of VERSION_GREATER with "."

Reply:
On dev4, running `./build.sh --use_cuda --use_full_protobuf --config Debug --build_wheel --enable_pybind --enable_training --nccl_home /bert_ort/wechi/nccl/nccl-2.7.3-1/build --parallel` produces
```
-- NCCL_MAJOR_VERSION: 2
-- NCCL_MINOR_VERSION: 7
-- NCCL P2P is enabled for supporting ncclSend and ncclRecv.
-- NCCL (include: /bert_ort/wechi/nccl/nccl-2.7.3-1/build/include, library: /bert_ort/wechi/nccl/nccl-2.7.3-1/build/lib/libnccl.so)
```
while `./build.sh --use_cuda --use_full_protobuf --config Debug --build_wheel --enable_pybind --enable_training --parallel` generates
```
-- NCCL_MAJOR_VERSION: 2
-- NCCL_MINOR_VERSION: 4
-- NCCL (include: /usr/include, library: /usr/lib/x86_64-linux-gnu/libnccl.so)
-- NCCL is enabled in Linux GPU Build.
```
This concludes the effectiveness of your suggestion.

On the other hand, I just add a check to only run this when version number is defined.